### PR TITLE
Small features

### PR DESCRIPTION
**Pull request by JureZmrzlikar**

### CODE REVIEW

File: resdk/resources/base.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ class BaseResource(object):
     endpoint = None
     query_endpoint = None
     query_method = 'GET'
+    full_search_keyword = None
```
Comment (dblenkus) [inline resdk/resources/base.py]: Rename to `full_search_paramater` as this is query parameter in url.

**Comment by JureZmrzlikar** [comment]
@dblenkus I've addressed you comments. Please make another review pass.

**Comment by JureZmrzlikar** [comment]
@dblenkus Please have another review pass when you find some time.

File: resdk/resources/base.py [hunk 2]
```diff
@@ -235,6 +232,20 @@ def contributor(self):
 
         return self._contributor
 
+    @property
+    def created(self):
+        """Creation time."""
+        if 'created' not in self._original_values:
+            raise ValueError('Instance does not have `created` attribute.')
```
Comment (dblenkus) [inline resdk/resources/base.py]: Can you add this only to applicable classes?

File: resdk/resources/data.py [hunk 3]
```diff
@@ -201,6 +199,20 @@ def descriptor_schema(self, dschema):
         # Save descriptor schema if already hydrated, otherwise it will be rerived in getter
         self._hydrated_descriptor_schema = dschema if is_descriptor_schema(dschema) else None
 
+    @property
+    def started(self):
+        """Start time."""
+        if 'started' not in self._original_values:
+            raise ValueError('Instance does not have `started` attribute.')
```
Comment (dblenkus) [inline resdk/resources/data.py]: This is not needed.

Comment (JureZmrzlikar) [reply resdk/resources/data.py]: Why not? `resdk.resources.Data` can be created locally and no `started` value will exist in `_original_values`. Do we return `None` or do we raise an error? Same goes for all: `created`, `modified`, `started`, `finished`.

File: resdk/resources/base.py (hunk 2 above)
Comment (JureZmrzlikar) [reply resdk/resources/base.py]: What do you mean? The only classes that inherit from this base class are those that have `created` and `modified` attributes...

File: resdk/resources/data.py (hunk 3 above)
Comment (dblenkus) [reply resdk/resources/data.py]: I see. The usual approach that we use for such cases is `if not self.id: raise`.

**Comment by JureZmrzlikar** [comment]
@dblenkus Comments addressed.

File: resdk/analysis/prepare_geo.py [hunk 4]
```diff
@@ -43,8 +43,7 @@ def prepare_geo_chipseq(resource, name=None):
 
     """
     reads = []
-    macs14 = []
-    relations = []
+    macs = []
```
Comment (dblenkus) [inline resdk/analysis/prepare_geo.py]: Don't forget to add a change log.

Comment (JureZmrzlikar) [reply resdk/analysis/prepare_geo.py]: This was just a "technical fix". The sinature of the function `prepare_geo_chipseq` did not change. So for the end user, there is no visible difference. AFAIK such changes don't belong to changelog...

Comment (dblenkus) [reply resdk/analysis/prepare_geo.py]: Ok, it makes sense.

**Pull request merged**